vignettes/customize.Rmd

```diff
@@ -7,40 +7,257 @@ vignette: >
   comment = "#>"
 )
-```
 
-```{r setup}
+# Setting up a tmp folder to use as the working directory for
+# the vignette. Also create a folder used in examples below.
+
+tmp <- withr::local_tempdir()
+dir.create(path = file.path(tmp, "my_root_path", "my_project"), recursive = TRUE)
+knitr::opts_knit$set(root.dir = tmp)
+
 library(connector)
 ```
 
-TODO: This is a vignette describing how to use the s3 dispatch to create your own 
-methods, and how to create your own connector classes.
-
-```{r, eval = FALSE}
-# Description from old connector_methods docs. To be rewritten for this vignette:
-#' Defaults methods for all connector object
-#'
-#' @description
-#' These methods are a S3 method that dispatches to the specific methods for the connector object.
-#'
-#' @details
-#' For example, if you have a connector object `connector_fs`, you can use `read_cnt(connector_fs, "file.csv")`
-#' to read the file. It will be dispatch to the `read_cnt.connector_fs` method.
-#' Why? The main aim is to allow the user to use the same function for different connector objects by
-#' using the builder function `connector_fs` and add an "extra_class".
-#' By doing so, you can create a subclass of the `connector_fs` object and
-#' dispatch to the specific methods for this subclass.
-#' For example, if you have a subclass `subclass`, you can use `cnr_read(subclass, "file.csv")` to read the file.
-#' It will be dispatch to the `read_cnt.subclass` method. And you can still use the `connector_fs` methods.
-#'
-#' @param connector_object A connector object to be able to use functions from it
-#' @param ... Additional arguments passed to the method
-#' @name connector_methods
-NULL
+`connector` is build to be easily extensible. This vignette will show you how to create your own connector classes, 
+and how to customize the existing ones, and their methods.
+
+The basic `connector` class is a simple `{R6}` object, with no additional functionality.
+It serves as the foundation for all other connectors, that should inherit from it in their `R6::R6Class()` definition -
+either directly or indirectly. Both `connector_fs()` and `connector_dbi()` inherit from `connector` in this way.
+
+## Creating a new connector
+
+We can create a new connector by creating a new R6 class that inherits from `connector`:
+
+```{r}
+connector_myclass <- R6::R6Class(
+  "connector_myclass",
+  inherit = connector
+)
+
+connector_myclass$new()
+```
+
+This is the simplest type of inheritance, and you should note that the `connector` 
+parent class has no methods for e.g. reading (`read_cnt()`) or writing (`write_cnt()`) data.
+
+In most cases you want to inherit from either `connector_fs()` or `connector_dbi()` depending
+on if your new connector is used to access files or databases respectively.
+
+Below we create a new `connector_project` class that inherits and acts exactly as the `connector_fs()`,
+but instead of the user having the provide the `path` to the folder as an argument, 
```

Comment:
small typo, it is "to provide" instead of "the provide"

Reply:
fixed